### ISSUE TITLE
Further course restructuring

### ISSUE THREAD

**Issue by sigilante** [urbit/developers.urbit.org#285]
Now that we have the framework for courses, I'd like to start mirroring course lessons publicly on this site.

![](https://raw.githubusercontent.com/sigilante/pixiesticks/master/school-design-20230119.png)

We currently have the Hoon School written docs which more or less correspond to the Hoon School Live lessons.  They're close enough that I wouldn't worry about mirroring HSL content.

App School Live content does deviate a fair bit from the App School guides, and I'd like to talk about what the best way to include those would look like.

- One option is to rename the App School/App School Full-Stack guides, but I don't have a good candidate for a new designation yet.  Plus they are the underlying reading content for ASL even if ASL handles it differently.

- Another option is to simply also include ASL material, but then where?

(Core School material will face a similar problem.  App Workshop material will probably all live in the App Workbook so it may be less of an issue.)

What do you think will work organizationally and name-wise?

@gordoncc @tinnus-napbus

**Comment by sigilante** [comment]
Current ASL content is at https://github.com/sigilante/curriculum/blob/master/asl-2023.1/

**Comment by sigilante** [comment]
Maybe slot school things under Courses instead of Guides?

**Comment by gordoncc** [comment]
Yeah, I'd say that new lessons should go under the course in /courses.  With others like HSL you can just link to the Hoon School docs.

**Comment by sigilante** [comment]
The top of every course lesson has a header like this:

```
---
title: "Agents"
teaching: 60
exercises: 0
nodes: []
objectives:
  - "Produce an agent which accepts and emits pokes and peeks."
  - "Produce an agent which accepts and handles subscriptions for a dataflow-reactive process."
runes: []
keypoints:
  - "A ⛓️Gall agent acts as a back-end persistent server process if it doesn't have a front-end interface."
  - "Agents communicate using `card`s."
  - "Pokes and peeks are for one-off checks; subscriptions are for monitoring and responding to changes in upstream agent data."
readings:
  - "https://developers.urbit.org/guides/core/app-school/5-cards"
  - "https://developers.urbit.org/guides/core/app-school/6-pokes"
  - "https://developers.urbit.org/guides/core/app-school/8-subscriptions"
  - "https://developers.urbit.org/guides/core/app-school/10-scry"
homework:
  - "https://forms.gle/8VcUbce1Dcca2qnBA"
mirror: "https://github.com/sigilante/curriculum/blob/master/asl-2023.1/asl2.md"
video: ""
---
```

How hard would it be to take that and render relevant info in a nice way?  CC @matildepark 

Not all are necessary but it'd be a pain to take them out of every file.  I'd want it to show:

- title
- objectives
- maybe runes
- keypoints (preferably at the _end_ of the page)
- readings (unless I just incorporate that back into the text)
- homework
- video, if any

**Comment by gordoncc** [comment]
@sigilante 

We can show whatever we want in the TOML, we'd just have to reformat the lists into proper arrays.  e.g:

runes: ["tiswut", "barcol", "hepcen"]

I'll do a pass on design first and we can review and then get it coded.
